packages/zowe-explorer/src/trees/dataset/DatasetTree.ts

```diff
@@ -1442,6 +1454,60 @@ export class DatasetTree extends ZoweTreeProvider<IZoweDatasetTreeNode> implemen
         return existing;
     }
 
+    /**
+     * Focuses on a specific data set in the dataset tree based on the current state of the tree.
+     *
+     * If the data set is not currently in the data sets filter, first check if it is in favourites.
+     * If it is, open that one, if not, fall back to setting the filter to that data set and then open it in the tree.
+     * Do not add filter to history.
+     *
+     * @param dsName - The name of the data set to focus on
+     * @param sessProfile - The session profile to use for the operation
+     * @returns {boolean} - True if the data set was found and focused
+     */
+    public async focusOnDsInTree(dsName: string, sessProfile: imperative.IProfileLoaded): Promise<boolean> {
+        // Try to find in session nodes
+        for (const session of this.mSessionNodes) {
+            const children = await session.getChildren();
+            const foundNode = children.find((child) => child.label?.toString().toUpperCase() === dsName.toUpperCase());
+            if (foundNode) {
+                await this.getTreeView().reveal(foundNode, { select: true, focus: true, expand: true });
+                return true;
+            }
+        }
+
+        // Try to find in favorites
+        for (const favNode of this.mFavorites) {
+            if (favNode.children && favNode.children.length > 0) {
+                const foundNode = favNode.children.find((child) => child.label?.toString().toUpperCase() === dsName.toUpperCase());
+                if (foundNode) {
+                    // Cannot just reveal foundNode as it will not expand out fully
+                    await this.getTreeView().reveal(favNode, { expand: true });
+                    await this.getTreeView().reveal(foundNode, { select: true, focus: true, expand: true });
+                    return true;
+                }
```

Comment:
I'm wondering if this comment can be resolved by passing a number to expand? See here: https://code.visualstudio.com/api/references/vscode-api#TreeDataProvider

If we pass a number (3 being the max value), it expands recursively up to the given number of times. That might help with it not expanding fully